lib/working_hours/computation.rb

```diff
@@ -246,5 +250,23 @@ def convert_to_original_format time, original
       end
     end
 
+    def holiday_hour?(time, config: nil)
+      matched_hours = []
+      config[:holiday_hours].each do |date, hours|
+        next unless time.to_date.to_s == date
+  
+        hours.each do |start_time, end_time|
+          if hour_to_seconds(time) >= start_time && hour_to_seconds(time) <= end_time
```

Comment:
I believe this `hour_to_seconds` method is the `time.seconds_since_midnight` method from ActiveSupport we use in other places in this project so you can probably use that instead to keep the code consistent.
Also you should only do this conversion one instead of N times inside the loop.
Finally, you should have the same time interval behavior we apply in `in_working_hours?` which means the end_time should be excluded.
Overall I would say you better start from the `in_working_hours?` method and modify it for your needs here as this one is quite different and less efficient.

Reply:
Killed the whole method now, `in_working_hours?` already takes into account any `holiday_hours` overridden for the day with `(config[:holiday_hours][time.to_date] || config[:working_hours][time.wday])`